### DISCUSSION TITLE
Chitchatter not working on both Librewolf and Ungoogled Chromium browsers.

### DISCUSSION THREAD

**Discussion by vicNric** [jeremyckahn/chitchatter#314] [Q&A]
hi, 
i've not been able to get chitchat working on either librewolf or ungoogled chromium. 
aware it may not work on such hardened browsers. but keen to know if i can change my settings so it does work. not keen on installing another browser just for video calls. 

i've attached some screenshots from mostly ungoog chromium, as well as one screenshot of librewolf 'partial network connection' error. 

on librewolf i've made sure the WebRTC setting needed is enabled (in the about:config settings). 
and i've taken a screenshot of the similar settings for chrome... but not 100% if what i have set up could be impacting it. 
just felt to ask. 

both test screenshots are just me opening chat, on my own, noone else connecting. 

if you's see or know of any tweaks i can make, appreciate it. 
or if think it's not possible in these browsers, i'll look at my options. 

thanks so much. really hope i can get this great resource to work for me. 

![Librewolf,partial connection](https://github.com/jeremyckahn/chitchatter/assets/109947572/69e6fd62-805b-4a1b-8993-7f503d7c3aba)

These three are all for ungoogled chromium.

![UngoogChrom-peers,no connection](https://github.com/jeremyckahn/chitchatter/assets/109947572/8ef76ae4-48cd-4fc3-a218-d595ce88d45b)

i'm not sure if i *enabled* this setting when set up. happy to disable if could be the issue. 
![webRTC search](https://github.com/jeremyckahn/chitchatter/assets/109947572/b2ba8271-d664-43db-bffd-33ade94c88fd)

i just added chitchatter into this field, to see if could be related. 

![about:flags(chrome settings)](https://github.com/jeremyckahn/chitchatter/assets/109947572/75fce597-3b5f-4aaa-9524-998813e458fb)

**Comment by jeremyckahn** [comment]
Hi @kaipie9, thanks for reporting this and for all the detailed information. The relay server message you're seeing in Librewolf suggests that the TURN server is having availability issues. This seems to affect other browsers right now as well, so I'm thinking there's an issue with the free TURN server that Chitchatter is currently using (see #112). We'll need to wait for a community member to provide server resources for this, as I cannot host this server myself.

Ungoogled Chromium doesn't seem to work at all for me. The red "No network connection" indicator means that the WebTorrent pairing servers can't be reached, but I can see in the devtools network monitor that messages are being passed back and forth as expected. It seems that Ungoogled Chromium is blocking the WebTorrent server in some way that's not expected and not being handled.

I don't know when I'll have time to get to the bottom of that issue, ~so I'll leave this issue open if others are able to find a fix~ but I'd welcome support from the community to address it.

**Reply by vicNric** [reply]
hi @jeremyckahn, 
thanks for the thorough reply and clarity on the situation. i read through the issue thread related to TURN server. i hope something works out in that regard, as seems like it's been ongoing for quite some time... 
seems ungogld chrom is not an option, but librewolf potentially? that if i try to use chitchat when a server is available it will work? 
any browsers you've found that work all the time? does feel bit much to install another browser just for video calls... but my current jitsi server i was using also didn't connect last time, so expect a less hardened browser would work for all calls (but then will be compromising on privacy and security, which goes against intentions on using software like chitchatter!). 

any suggestions appreciated. if doesn't work, will just have to use jitsi logging in with github account, or if you know of reliable instance or other option, appreciate hearing. 
thanks for your great work.

**Reply by jeremyckahn** [reply]
Hi @kaipie9, it seems that LibreWolf should work about as well as anything for basic chatting. Any connectivity issues you experience are likely due to TURN server availability/flakiness that would affect all browsers. Also, it can take a moment for peers to connect even in normal circumstances due to the P2P nature of the app.

> any browsers you've found that work all the time?

I generally use Firefox and Chromium for most things so I know those will fully work with Chitchatter. I'm noticing that video calls aren't working with LibreWolf because it appears to be blocking camera access to web apps. I'm not sure if that's something that can be configured within LibreWolf, but it doesn't seem like it can be addressed at the application level. Voice calls seem to work, so that's something.

There's only so much that can be done at the application level within the constraints of a hardened browser. If you want full functionality, you'll likely have to switch to a more mainstream browser. Sorry I can't give you a better answer than that!

**Reply by vicNric** [reply]
thanks jeremy for the clarity. good to know all this. 
since last msg i've found a few solid jitsi instances, which i feel i'll use for now... as not keen on setting up and storing another browser on my system, with LW not seemingly suiting chitchatter, presently. 
again, appreciate the support. hope you guys get sorted around the TURN server situation... if so, i'll gladly start using chitchatter.
